packages/@aws-cdk/aws-vpcv2-alpha/lib/ipam.ts

```diff
@@ -0,0 +1,541 @@
+
+  /**
+   * IPAM resource name
+   * @default autogenerated by CDK
+   */
+  readonly ipamPoolName?: string;
+}
+
+/**
+ * Options to provision CIDRs to an IPAM pool.
+ * Used to create a new IpamPoolCidr
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampoolcidr.html
+ */
+export interface IpamPoolCidrProvisioningOptions {
+  /**
+   * Ipv6 Netmask length for the CIDR
+   * @default none
+   */
+  readonly netmaskLength?: number;
+
+  /**
+   * Ipv6 CIDR block for the IPAM pool
+   * @default none
+   */
+  readonly cidr?: string;
+}
+
+/**
+ * Definition used to add or create a new IPAM pool
+ */
+export interface IIpamPool{
+  /**
+ * Pool ID to be passed to the VPC construct
+ * @attribute IpamPoolId
+ */
+  readonly ipamPoolId: string;
+
+  /**
+   * Pool CIDR for IPv6 to be provisioned with Public IP source set to 'Amazon'
+   */
+  readonly ipamCidrs: CfnIPAMPoolCidr[];
+
+  /**
+   * Function to associate a IPv6 address with IPAM pool
+   */
+  provisionCidr(id: string, options: IpamPoolCidrProvisioningOptions): CfnIPAMPoolCidr;
+
+}
+
+/**
+ * IPAM scope is the highest-level container within IPAM. An IPAM contains two default scopes.
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipamscope.html
+ */
+export interface IpamScopeProps extends IpamScopeOptions{
+  /**
+   * IPAM id to which scope needs to be added
+   */
+  readonly ipamId: string;
+
+  /**
+   * Operating regions for the Ipam
+   * Required in order to validate the locale being set on pool
+   */
+  readonly ipamOperatingRegions: string[];
+
+  /**
+   * Custom ipam scope id to add a pool in order to support default scopes
+   * @default none
+   */
+  readonly ipamScopeId?: string;
+
+}
+
+/**
+ * Being used in IPAM class to add pools to default scope created by IPAM.
+ */
+export interface IpamScopeOptions {
+
+  /**
+   * IPAM scope name that will be used for tagging
+   * @default none
+   */
+  readonly ipamScopeName?: string;
+}
+
+/**
+ * Options for configuring an IP Address Manager (IPAM).
+ *
+ * For more information, see the {@link https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipam.html}.
+ */
+export interface IpamOptions {
+
+  /**
+   * CIDR Mask for Vpc
+   * Only required when using AWS Ipam
+   *
+   * @default - None
+   */
+  readonly netmaskLength?: number;
+
+  /**
+   * Ipv4 or an Ipv6 IPAM pool
+   * Only required when using AWS Ipam
+   *
+   * @default - None
+   */
+  readonly ipamPool?: IIpamPool;
+}
+
+/**
+ * Interface for IpamScope Class
+ */
+export interface IIpamScopeBase{
+
+  /**
+   * Reference to the current scope of stack to be passed in order to create
+   * a new IPAM pool
+   */
+  readonly scope: Construct;
+
+  /**
+   * Default Scope ids created by the IPAM or a new Resource id
+   */
+  readonly scopeId: string;
+
+  /**
+   * Properties to configure IPAMScope
+   */
+  readonly props?: IpamScopeProps;
```

Comment:
the intent to make `IpamScopeProps`, so I think this property should be of type `IpamScopeOptions` instead.

Reply:
or we can make it a private property